lib/matplotlib/patches.py

```diff
@@ -425,7 +425,7 @@ def set_color(self, c):
 
     def _set_hatchcolor(self, color):
         color = mpl._val_or_rc(color, 'hatch.color')
-        if color == 'edge':
+        if isinstance(color, str) and color == 'edge':
```

Comment:
Below in `set_hatchcolor` we call _str_equal and then set the color to the value we checked against which doesn't make sense to me. I think that call was likely supposed to go up here. So we could remove the check in the below function as well.
```suggestion
        if cbook._str_equal(color, 'edge'):
```